org.eclipse.wildwebdeveloper.tests/src/org/eclipse/wildwebdeveloper/tests/TestLanguageServers.java

```diff
@@ -233,5 +248,70 @@ protected boolean condition() {
 			}
 		}.waitForCondition(PlatformUI.getWorkbench().getDisplay(), 3000));
 	}
+	
+	@Test
+	public void testAngularTSFile() throws Exception {
+		URL url = FileLocator.find(Platform.getBundle("org.eclipse.wildwebdeveloper.tests"), Path.fromPortableString("testProjects/angular-app"), null);
+		url = FileLocator.toFileURL(url);
+		File folder = new File(url.getFile());
+		if (folder.exists()) {
+			FileUtils.copyDirectory(folder, project.getLocation().toFile());
+			Process process = new ProcessBuilder(getNpmLocation(), "install", "--no-bin-links", "--ignore-scripts").directory(project.getLocation().toFile()).start();
+			while (process.isAlive());
+			project.refreshLocal(IResource.DEPTH_INFINITE, new NullProgressMonitor());			
+			IFile file = project.getFolder("src").getFolder("app").getFile("app.component.ts");
+			String fileContent = IOUtils.toString(file.getContents(), StandardCharsets.UTF_8);
+			ITextEditor editor = (ITextEditor) IDE.openEditor(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage(), file);
+			editor.getDocumentProvider().getDocument(editor.getEditorInput()).set(fileContent);
```

Comment:
Does that line change anything? It's setting document with same contents as previously. Is it "only" to trigger an event?
If so, maybe it'd be best if you start from a clean file, check there are not markers and then make a modification on the document and see that markers are added accordingly.